### PR TITLE
Droth 609 vvh client refactory

### PR DESCRIPTION
**Pull request by leirasf**
Just to have a look to see if it looks good then we will need to merge it with the DROTH-609. And reorganize it in different files.

### CODE REVIEW

File: digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/AssetDataImporterSpec.scala [hunk 1]
```diff
@@ -392,7 +392,7 @@ class AssetDataImporterSpec extends FunSuite with Matchers {
   */
 
   //case 1
-  test("Should unfloat the obstacle when exists just one roadlink inside a radius of 10 meters"){
+  test("Should unfloat the obstacle when exists just one roadliDo not overwrite LiviId of ELY/HSL stops when Tietojen ylläpitäjä is empty in csv import filenk inside a radius of 10 meters"){
```
Comment (halied) [inline digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/AssetDataImporterSpec.scala]: What has happened to this comment?

**Comment by halied** [comment]
Is something going to be done to those todo comments? Otherwise ok.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/VVHClient.scala [hunk 2]
```diff
@@ -298,250 +310,226 @@ class VVHClient(vvhRestApiEndPoint: String) {
...
-  
-  def queryChangesByPolygon(polygon: Polygon): Seq[ChangeInfo] = {
-    val polygonString = stringifyPolygonGeometry(polygon)
-    if (!polygonString.contains("{rings:["))
-      return  Seq.empty[ChangeInfo]
-    val defs="[{\"layerId\":0,\"outFields\":\"OLD_ID,NEW_ID,MTKID,CHANGETYPE,OLD_START,OLD_END,NEW_START,NEW_END,CREATED_DATE,CONSTRUCTIONTYPE\"}]"
-    val url = vvhRestApiEndPoint + "/Roadlink_ChangeInfo/FeatureServer/query?" +
-    "layerDefs="+URLEncoder.encode(defs, "UTF-8")+"&geometry=" +  URLEncoder.encode(polygonString) + "&geometryType=esriGeometryPolygon&spatialRel=esriSpatialRelIntersects&returnGeometry=false&outFields=false&f=pjson"
+
+  //TODO maybe this should go to roadlink
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/VVHClient.scala]: Move this method to VVHRoadlinkClient

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/VVHClient.scala [hunk 3]
```diff
@@ -298,250 +310,226 @@ class VVHClient(vvhRestApiEndPoint: String) {
...
+  protected val featureClassCodeToFeatureClass: Map[Int, FeatureClass] = Map(
+    12316 -> FeatureClass.TractorRoad,
+    12141 -> FeatureClass.DrivePath,
+    12314 -> FeatureClass.CycleOrPedestrianPath
+  )
+
+  //TODO maybe this should go to roadlink
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/VVHClient.scala]: Move this method to VVHRoadlinkClient

**Comment by joanacoelho** [comment]
Looks good

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/VVHClient.scala [hunk 4]
```diff
@@ -767,281 +694,297 @@ class VVHClient(vvhRestApiEndPoint: String) {
...
+  protected override def defaultOutFields(): String = {
+    "OBJECTID,NODEID,FORMOFNODE,MUNICIPALITYCODE"
+  }
+
+  protected override def mapFields(content: Map[String, Any], url: String): Either[List[Map[String, Any]], VVHError] = {
+    val optionalLayers = content.get("layers").map(_.asInstanceOf[List[Map[String, Any]]])
+    val optionalFeatureLayer = optionalLayers.flatMap { layers => layers.find { layer => layer.contains("features") } }
+    val optionalFeatures = optionalFeatureLayer.flatMap { featureLayer => featureLayer.get("features").map(_.asInstanceOf[List[Map[String, Any]]]) }
+    //TODO Check if the change info have contruction type
+    //optionalFeatures.map(_.filter(roadLinkStatusFilter)).map(Left(_)).getOrElse(Right(VVHError(content, url)))
```
Comment (marquesfm) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/VVHClient.scala]: This comment it's for remove? Or is related with the TODO?

**Comment by marquesfm** [comment]
ok for me

**Comment by tvenhola** [comment]
seal-of-approval-meme

**Pull request merged**